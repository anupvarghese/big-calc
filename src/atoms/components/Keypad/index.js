// @flow

import React from 'react';
import { css } from 'glamor';
import C from '../../../constants';

const baseKeypadStyle = {
  display: 'flex',
  flexDirection: 'column',
  background: C.GRAY_BROWN,
  borderTopColor: C.BLACK,
};

const keyRowStyle = {
  display: 'flex',
  width: '100%',
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
  textAlign: 'center',
};

const keyStyle = (operator = false) => {
  const color = operator ? C.ORANGE : C.GRAY_WHITE;
  return {
    flex: 1,
    margin: '0.5px',
    border: `2px solid ${color}`,
    padding: '1rem',
    background: color,
    fontWeight: 'medium',
    fontSize: '1.3rem',
  };
};

type rowT = {
  style: ?Object,
  children: React.Node,
};

type keyT = {
  style: ?Object,
  handleOnClick: () => {},
  tag: string,
};

type keypadT = {
  keypadStyle: ?Object,
  handleNumberClick: () => {},
  handleOperatorClick: () => {},
};

const Row = ({ style, children }: rowT) => (
  <div {...css(keyRowStyle, style)}>{children}</div>
);
const Key = ({ style, handleOnClick, tag }: keyT) => (
  <div
    {...css(keyStyle(['/', 'X', '-', '+', '='].includes(tag)), style)}
    onClick={handleOnClick(tag)}
  >
    {tag}
  </div>
);

const Keypad = ({
  keypadStyle,
  handleNumberClick,
  handleOperatorClick,
}: keypadT) => {
  return (
    <div {...css(baseKeypadStyle, keypadStyle)}>
      <Row>
        {['AC', '+/-', '%', '/'].map(k => (
          <Key key={k} tag={k} handleOnClick={handleOperatorClick} />
        ))}
      </Row>
      <Row>
        {['7', '8', '9'].map(k => (
          <Key key={k} tag={k} handleOnClick={handleNumberClick} />
        ))}
        <Key key="X" tag="X" handleOnClick={handleOperatorClick} />
      </Row>
      <Row>
        {['4', '5', '6'].map(k => (
          <Key key={k} tag={k} handleOnClick={handleNumberClick} />
        ))}
        <Key key="-" tag="-" handleOnClick={handleOperatorClick} />
      </Row>
      <Row>
        {['1', '2', '3'].map(k => (
          <Key key={k} tag={k} handleOnClick={handleNumberClick} />
        ))}
        <Key key="+" tag="+" handleOnClick={handleOperatorClick} />
      </Row>
      <Row>
        <div {...css({ flex: 2 })}>
          <Key key="0" tag="0" handleOnClick={handleNumberClick} />
        </div>
        <div {...css({ display: 'flex', flexDirection: 'row', flex: 2 })}>
          <Key key="." tag="." handleOnClick={handleNumberClick} />
          <Key key="=" tag="=" handleOnClick={handleOperatorClick} />
        </div>
      </Row>
    </div>
  );
};

export default Keypad;
