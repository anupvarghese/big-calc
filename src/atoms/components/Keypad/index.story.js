import React from 'react';

import { storiesOf } from '@storybook/react';
import Keypad from './';

storiesOf('Keypad').add('show keypad', () => (
  <Keypad keypadStyle={{ width: '400px' }} handleKeyClick={() => () => {}} />
));
