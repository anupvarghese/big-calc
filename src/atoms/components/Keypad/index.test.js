import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-glamor-react';

import Keypad from './';

expect.addSnapshotSerializer(serializer);

describe('Keypad', () => {
  it('renders Keypad', () => {
    const keypad = renderer
      .create(
        <Keypad handleNumberClick={() => {}} handleOperatorClick={() => {}} />,
      )
      .toJSON();
    expect(keypad).toMatchSnapshot();
  });
});
