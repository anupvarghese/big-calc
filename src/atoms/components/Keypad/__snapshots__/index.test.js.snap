// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad renders Keypad 1`] = `
.css-tcqo1y,
[data-css-tcqo1y] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  border: 1px solid tomato;
  border-radius: 4px;
  margin-top: 0.5rem;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-1qeez79,
[data-css-1qeez79] {
  flex: 2;
  -webkit-flex: 2;
}

.css-q8ffom,
[data-css-q8ffom] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  flex: 2;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-flex: 2;
}

.css-h0wfqv,
[data-css-h0wfqv] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-zcaaxo,
[data-css-zcaaxo] {
  flex: 1;
  margin: 0.5rem;
  border: 1px solid tomato;
  padding: 1rem;
  -webkit-flex: 1;
}

<div
  data-css-tcqo1y=""
>
  <div
    data-css-h0wfqv=""
  >
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      AC
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      +/-
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      %
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      /
    </div>
  </div>
  <div
    data-css-h0wfqv=""
  >
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      7
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      8
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      9
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      X
    </div>
  </div>
  <div
    data-css-h0wfqv=""
  >
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      4
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      5
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      6
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      -
    </div>
  </div>
  <div
    data-css-h0wfqv=""
  >
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      1
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      2
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      3
    </div>
    <div
      data-css-zcaaxo=""
      onClick={undefined}
    >
      +
    </div>
  </div>
  <div
    data-css-h0wfqv=""
  >
    <div
      data-css-1qeez79=""
    >
      <div
        data-css-zcaaxo=""
        onClick={undefined}
      >
        0
      </div>
    </div>
    <div
      data-css-q8ffom=""
    >
      <div
        data-css-zcaaxo=""
        onClick={undefined}
      >
        .
      </div>
      <div
        data-css-zcaaxo=""
        onClick={undefined}
      >
        =
      </div>
    </div>
  </div>
</div>
`;
